dev/check_patch_prs.py

```diff
@@ -11,54 +12,80 @@ def get_release_branch(version):
     return f"branch-{major_minor_version}"
 
 
-def get_merged_prs(version):
+@dataclass(frozen=True)
+class Commit:
+    sha: str
+    pr_num: int
+
+
+def get_commits(branch: str):
     """
-    Get the PRs merged into the release branch for the given version.
+    Get the commits in the release branch.
     """
-    release_branch = get_release_branch(version)
-    commits = subprocess.check_output(
+    log_stdout = subprocess.check_output(
         [
             "git",
             "log",
-            release_branch,
-            '--since="2 months ago"',
-            "--pretty=format:%s",
+            branch,
+            '--since="3 months ago"',
+            "--pretty=format:%H %s",
         ],
         text=True,
     )
-    pr_rgx = re.compile(r"\s+\(#(\d+)\)$")
-    prs = set()
-    for commit in commits.splitlines():
-        if pr := pr_rgx.search(commit.rstrip()):
-            prs.add(int(pr.group(1)))
+    pr_rgx = re.compile(r"([a-z0-9]+) .+\s+\(#(\d+)\)$")
+    commits = []
+    for commit in log_stdout.splitlines():
+        if m := pr_rgx.search(commit.rstrip()):
+            commits.append(Commit(sha=m.group(1), pr_num=int(m.group(2))))
+
+    return commits
+
 
-    return prs
+@dataclass(frozen=True)
+class PR:
+    pr_num: int
+    merged: bool
 
 
 def fetch_patch_prs(version):
     """
     Fetch PRs labeled with `patch-{version}` from the MLflow repository.
     """
     label = f"patch-{version}"
-    response = requests.get(
-        f'https://api.github.com/search/issues?q=is:pr+repo:mlflow/mlflow+label:"{label}"'
-    )
-    response.raise_for_status()
-    data = response.json()
-    return {pr["number"] for pr in data["items"]}
+    per_page = 100
+    page = 1
+    pulls = []
+    while True:
+        response = requests.get(
+            f'https://api.github.com/search/issues?q=is:pr+repo:mlflow/mlflow+label:"{label}"&per_page={per_page}&page={page}',
+        )
+        response.raise_for_status()
+        data = response.json()
+        pulls.extend(data["items"])
+        if len(data) < per_page:
+            break
+        page += 1
+
+    return {pr["number"]: pr["pull_request"].get("merged_at") is not None for pr in data["items"]}
```

Comment:
```suggestion
    return {pr["number"]: pr["pull_request"].get("merged_at") is not None for pr in pulls}
```
I guess you want to use `pulls` here

Reply:
good catch!